### PR TITLE
implemented retrievement of id after commit()

### LINKED ISSUE #484: Set document ID after commit to ES
```php
$document = new CustomDocument();

$manager->persist($document);
$manager->commit();

var_dump($document->getId());
```

Currently this code returns `NULL`. It would be nice to get generated ID.

### PR DESCRIPTION
**Pull request by einorler**
closes #484

### CODE REVIEW

File: Service/Manager.php [hunk 1]
```diff
@@ -321,6 +332,38 @@ public function refresh(array $params = [])
...
+            if (isset($query['index']) && !isset($query['index']['_id'])) {
+                $indexing[] = $number / 2;
+            }
+        }
+
+        if (isset($bulkQueries['body'][0]['index'])) {
+            if (isset($this->persistedObjects)) {
+                $i = 0;
+                foreach ($this->persistedObjects as $document) {
+                    $document->id = $bulkResponse['items'][$indexing[$i]]['create']['_id'];
```
Comment (mvar) [inline Service/Manager.php]: You can not hardcode `id` property here. User is free to choose which property of a document he wants to use for Elasticsearch `_id` value (using `@Id` annotation).

You could get actual property from metadata collector.

**Comment by einorler** [comment]
Thanks for that, I've taken it into account in the latest changes

File: Service/Manager.php [hunk 2]
```diff
@@ -321,6 +332,44 @@ public function refresh(array $params = [])
...
+        if (isset($bulkQueries['body'][0]['index'])) {
+            if (isset($this->persistedObjects)) {
+                $i = 0;
+                foreach ($this->persistedObjects as $document) {
+                    $class = get_class($document);
+                    $mapping = $this->metadataCollector->getMapping($class);
+
+                    if (isset( $mapping['aliases']['_id'])) {
+                        $id_property = $mapping['aliases']['_id']['propertyName'];
+                        $document->$id_property = $bulkResponse['items'][$indexing[$i]]['create']['_id'];
```
Comment (mvar) [inline Service/Manager.php]: This property can be private :)

**Comment by saimaz** [comment]
It's been remastered in #660 with grater performance.

**Pull request closed without merging**